Description: Add BUFRNUM() function to bufrdc/ code. According to docs, each interface supplies
 A function to describe its version number, but bufrnum was not present.
Author: Alastair McKinstry <mckinstry@debian.org>
Forwarded: no
Last-Updated: 2015-05-13

# From gribex/emoscyc.F, it appears that each interface has its own _CYCLE number,
# emoscyc.F used BUFRNUM, but it was never defined; I've defined it here
# matching gribnum, intpnum, etc.
# -- AMCK, 2009-04-16
Index: emoslib-4.4.9/bufrdc_wmo/bufrnum.F
===================================================================
--- /dev/null
+++ emoslib-4.4.9/bufrdc_wmo/bufrnum.F
@@ -0,0 +1,122 @@
+C 
+C 
+C Licensed under the GNU Lesser General Public License which
+C incorporates the terms and conditions of version 3 of the GNU
+C General Public License.
+C See LICENSE and gpl-3.0.txt for details.
+C
+
+      INTEGER FUNCTION BUFRNUM(KONOFF)
+C
+C---->
+C**** BUFRNUM
+C
+C     Purpose
+C     -------
+C
+C     Returns current BUFR_CYCLE version number.
+C
+C
+C     Interface
+C     ---------
+C
+C     INUM = BUFRNUM(KONOFF)
+C
+C     Input
+C     -----
+C
+C     KONOFF - switch for displayed message
+C              = 0 if display is required on the first call
+C              = non-zero if display is not required
+C
+C
+C     Output
+C     ------
+C
+C     Returns a 6-digit version number, aaabbc, where:
+C       aaa = 3-digit major number
+C       bb  = 2-digit minor number
+C       c   = 1-digit spare number (normally 0)
+C     
+C
+C     Method
+C     ------
+C
+C     Reads a 6-digit version number from the environment variable
+C     BUFR_CYCLE. If this does not give a 6-digit number, an internal
+C     hard-coded default value is used.
+C
+C     On the first call, the function (optionally) displays a message:
+C
+C       **************************************
+C       * BUFR_CYCLE version number = aaabbc *
+C       **************************************
+C     
+C
+C     Externals
+C     ---------
+C
+C     None.
+C     
+C
+C     Author
+C     ------
+C
+C     A.McKinstry    ICHEC     Apr 2009
+C
+C
+C----<
+C ------------------------------------------------------------------
+C*    Section 0.   Variables.
+C ------------------------------------------------------------------
+C
+C
+      IMPLICIT NONE
+C
+#include "grprs.h"
+C
+C     Function arguments
+C
+      INTEGER KONOFF
+C
+C     Local variables
+C
+      INTEGER INUMBER, ICOUNT, IOFFSET
+      SAVE INUMBER, ICOUNT
+      CHARACTER*38 CMESS
+      CHARACTER*20 YNUMBER
+C
+      DATA INUMBER/000010/, ICOUNT/0/
+      DATA CMESS/'* BUFR_CYCLE version number = ****** *'/
+C
+C ------------------------------------------------------------------
+C*    Section 1.   Initialise
+C ------------------------------------------------------------------
+C
+  100 CONTINUE
+C
+      IF( ICOUNT.EQ.0 ) THEN
+C
+C       See if the environment variable has an override value
+C
+        CALL GETENV( 'BUFR_CYCLE', YNUMBER)
+        IOFFSET = INDEX( YNUMBER, ' ')
+        IF( IOFFSET.EQ.7 ) THEN
+          READ(YNUMBER,'(I6.6)') INUMBER
+        ENDIF
+C
+C       First time through, display the message if required
+C
+        IF( KONOFF.EQ.0 ) THEN
+          WRITE(CMESS(31:36),'(I6.6)') INUMBER
+          WRITE(GRPRSM,*) '**************************************'
+          WRITE(GRPRSM,*) CMESS
+          WRITE(GRPRSM,*) '**************************************'
+        ENDIF
+        ICOUNT = 1
+      ENDIF
+C
+      BUFRNUM = INUMBER
+C
+      RETURN
+      END
Index: emoslib-4.4.9/bufrdc_wmo/grprs.h
===================================================================
--- /dev/null
+++ emoslib-4.4.9/bufrdc_wmo/grprs.h
@@ -0,0 +1,7 @@
+C
+C     Common blocks holding default or user supplied values for printing.
+C
+      LOGICAL GRPRSET
+      INTEGER GRPRSM
+C
+      COMMON /GRPRSCM/ GRPRSM, GRPRSET
Index: emoslib-4.4.9/bufrdc_wmo/CMakeLists.txt
===================================================================
--- emoslib-4.4.9.orig/bufrdc_wmo/CMakeLists.txt
+++ emoslib-4.4.9/bufrdc_wmo/CMakeLists.txt
@@ -2,6 +2,7 @@
 #   Sources for bufrdc
 #
 list( APPEND bufrdc_wmo_src_files
+   bufrnum.F
    bbuprs0.F
    bbuprs1.F
    bbuprs2.F
